### PR TITLE
Task/74 user can use filters

### PR DESCRIPTION
**Pull request by Techies-John**
## Problem/Issue:
- Linked Issue: #74  

The Search Filter UI has undergone significant changes, requiring updates to: 
1. The factory (algorithm)
2. UI functions.

## Proposed Solution
1. Factory Refactor: Adjust and enhance the algorithm to align with the updated UI.
2. Function Refactor: Update and optimize UI functions to support the new filter behavior.

## Alternative Solutions
**Maintain Current Algorithm:**
- **Pros:** Avoids reliance on third-party systems, preserving full control.
- **Cons:** Enhancing the algorithm requires more time and effort, as it’s developed by a single contributor.

**Adopt Third-Party Algorithm:**
- **Pros:** Ready-made solution with potentially higher search accuracy.
- **Cons:**
   - Involves a learning curve and system adjustments.
   - Limits flexibility since the third-party solution might not cover all desired functionalities.

## Suggestions:
- UI Improvement: Replace the filter year field with a single slider to simplify user interaction.

## Summary
Users can now enjoy an upgraded filter experience! Yay🥳🥳🥳

### CODE REVIEW

File: components/search/area/SearchBar.vue [hunk 1]
```diff
@@ -95,11 +105,13 @@ function handleFilterDropdownState(isOpenValue: boolean) {
     isOpen.value = isOpenValue
 }
 
-function handleSelectedYear(selectedYearValue: string) {
+function handleSelectedYear(selectedYearValue: DateRangeType) {
+    console.log("selectedYearLower: " + selectedYear.value.lowerYear)
```
Comment (arkislove) [inline components/search/area/SearchBar.vue:109]: hala console.log

Comment (Techies-John) [reply components/search/area/SearchBar.vue:109]: haha yati kuyawa man paspasa maka kitag pr lupigan paning kauban

**Review by nonsensenv — APPROVED** [review]

**Comment by Techies-John** [comment]
I will now merge this branch so that other devs could now start doing their tasks

**Comment by arkislove** [comment]
okay

**Comment by arkislove** [comment]
> I will now merge this branch so that other devs could now start doing their tasks

okay

**Pull request merged**